### PR TITLE
Storage Doc Update for pluggable architecture

### PR DESCRIPTION
**Pull request by nidsharm**

### CODE REVIEW

File: js/storage.md [hunk 1]
```diff
@@ -61,13 +61,16 @@ Amplify.configure({
         userPoolWebClientId: 'XX-XXXX-X_abcd1234', //OPTIONAL - Amazon Cognito Web Client ID
     },
     Storage: {
-        bucket: '', //REQUIRED -  Amazon S3 bucket
-        region: 'XX-XXXX-X', //OPTIONAL -  Amazon service region
+        AWSS3: {
```
Comment (elorzafe) [inline js/storage.md]: You will need to update line 157 `Storage.configure`

File: js/storage.md [hunk 2]
```diff
@@ -61,13 +61,16 @@ Amplify.configure({
...
+            bucket: '', //REQUIRED -  Amazon S3 bucket
+            region: 'XX-XXXX-X', //OPTIONAL -  Amazon service region
+        }
     }
 });
 
 ```
 
+## Using Amazon S3
```
Comment (elorzafe) [inline js/storage.md]: I think we could add instructions for other xyz provider and how that will configure those as well on other providers section. Analytics has a section for that https://aws-amplify.github.io/docs/js/analytics#using-a-custom-plugin

File: js/storage.md (hunk 1 above)
Comment (nidsharm) [reply js/storage.md]: done!

File: js/storage.md (hunk 2 above)
Comment (nidsharm) [reply js/storage.md]: Added the custom plugin section.

File: js/storage.md [hunk 3]
```diff
@@ -345,6 +349,69 @@ Storage.list('photos/', {level: 'private'})
...
+// send configuration into Amplify
+Storage.configure({
+    MyStorageProvider: { 
+        // My Storage provider configuration 
+    }
+});
+
+```
+
+The default provider (Amazon S3) is in use when you call `Storage.put( )` unless you specify a different provider: `Storage.put(key, object, {'MyStorageProvider'})`.
```
Comment (elorzafe) [inline js/storage.md]:
```suggestion
The default provider (Amazon S3) is in use when you call `Storage.put( )` unless you specify a different provider: `Storage.put(key, object, { provider: 'MyStorageProvider'})`. 
```

File: js/storage.md [hunk 4]
```diff
@@ -345,6 +349,69 @@ Storage.list('photos/', {level: 'private'})
...
+
+// get the plugin
+Storage.getPluggable(MyStorageProvider.providerName);
+
+// remove the plulgin
+Storage.removePluggable(MyStorageProvider.providerName);
+
+// send configuration into Amplify
+Storage.configure({
+    MyStorageProvider: {
```
Comment (elorzafe) [inline js/storage.md]:
```suggestion
    [MyStorageProvider.providerName]: { 
```

File: js/storage.md [hunk 5]
```diff
@@ -149,12 +152,13 @@ Import *Storage* from the aws-amplify library:
...
 ```
 
-If you use `aws-exports.js` file, Storage is already configured. To configure Storage manually,
+If you use `aws-exports.js` file, Storage is already configured when you call `Amplify.configure(awsmobile)`. To configure Storage manually,
 ```javascript
-Storage.configure({
-    bucket: //Your bucket ARN;
-    region: //Specify the region your bucket was created in;
-    identityPoolId: //Specify your identityPoolId for Auth and Unauth access to your bucket;
+Storage.configure(
```
Comment (elorzafe) [inline js/storage.md]:
```
Storage.configure({
    AWSS3: {
        bucket: '', //Your bucket ARN
        region: '', //Specify the region your bucket was created in
     }
});
```

And I would add a comment that is required to configure Auth in order to S3 with credentials

**Pull request closed without merging**